sound/soc/sof/sof-client-ipc-kernel-injector.c

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: GPL-2.0-only
+//
+// Copyright(c) 2022 Google Inc. All rights reserved.
+//
+// Author: Curtis Malainey <cujomalainey@chromium.org>
+//
+
+#include <linux/auxiliary_bus.h>
+#include <linux/completion.h>
+#include <linux/debugfs.h>
+#include <linux/ktime.h>
+#include <linux/mod_devicetable.h>
+#include <linux/module.h>
+#include <linux/pm_runtime.h>
+#include <linux/slab.h>
+#include <linux/uaccess.h>
+#include <sound/sof/header.h>
+#include <sound/sof/ipc4/header.h>
```

Comment:
do you need this include? it doesnt look like IPC4 is supported by this client just yet

Reply:
Ah thanks for catching my copy paste error

Reply:
Actually this is needed, since the injector itself supports IPC4, it is the sof-client that does not, so I think I will leave it for the alloc modification

Reply:
@ujfalusi that is the point. The kernel should not trust the data and should reject malformed payloads. 

And yes there will be a IPC4 implementation. Regardless of semantics, we have security requirements to fuzz this interface.